harmony_regridding_service/grid.py

```diff
@@ -105,16 +105,68 @@ def same_source_and_target_crs(
 
 
 def create_target_area_from_source(
-    filepath: str, var_info: VarInfoFromNetCDF4, crs: CRS
+    filepath: str, var_info: VarInfoFromNetCDF4, target_crs: CRS
 ) -> AreaDefinition:
-    """Create the target area definition using the source grid information.
+    """Create the target area using the source grid information.
 
     TODO: Create area definition for every dimension pair, if the collection
     had more than one grid.
     """
     dimension_pairs = get_resampled_dimension_pairs(var_info)
-    return create_area_definition_for_projected_source_grid(
-        filepath, dimension_pairs[0], var_info, override_crs=crs
+    projected_area = create_area_definition_for_projected_source_grid(
+        filepath, dimension_pairs[0], var_info
+    )
+
+    # I have the correct projected areaDefinition, but I want to convert it to
+    # a geographic area
+    geographic_area = convert_projected_area_to_geographic(projected_area, target_crs)
+
+    return geographic_area
+
+
+def convert_projected_area_to_geographic(
+    projected_area: AreaDefinition, target_crs: CRS
+) -> AreaDefinition:
+    """Converts a Projected AreaDefinition into the similar area in the target_CRS.
+
+    For now the target_crs is always goign to be CRS('epsg:4326')
+
+    This is the "simple case" where we know that the x/y corners transform
+    directly to lon/lat corners. ***This does not handle polar grids.***
+    to handle polar you may be able to use area.boundary().
+
+    """
+    geographic_area = create_area_def(
+        'Geographic Area',
+        target_crs,
+        area_extent=reorder_extents(*projected_area.area_extent_ll),
+        width=projected_area.width,
+        height=projected_area.height,
+        shape=projected_area.shape,
+    )
+    logger.debug(f'Source projected Area: {projected_area}')
+    logger.debug(f'Converted Geographic Area: {geographic_area}')
+
+    return geographic_area
+
+
+def reorder_extents(min_x, min_y, max_x, max_y):
+    """This is a way to ensure the correct area extents.
+
+    # The pyresample generated area_extent_ll is not always returning the
+    # expected values for the area extent point documented as (lower_left_lon,
+    # lower_left_lat, upper_right_lon, upper_right_lat) Resulting in some bad
+    # resampling outputs. This may be a stop-gap/workaround?
```

Comment:
Is this potentially because input grids might not be aligned to have longitude or latitude increasing in the expected direction along the array axes? (I think one of the gridded ICESat-2 collections is upside-down for example) Or maybe also because the grid corners of a projected grid might not be the min/max lat/lon?

(It's probably not too important, as what you do to get the return value makes sense to me, I just was trying to think through why there's a discrepancy)